core/src/io.rs

```diff
@@ -93,14 +93,16 @@ pub struct MetaIoHandler<T: Metadata, S: Middleware<T> = middleware::Noop> {
 
 impl<T: Metadata> Default for MetaIoHandler<T> {
 	fn default() -> Self {
-		MetaIoHandler::with_compatibility(Default::default())
+		Self::with_compatibility(Compatibility::default())
 	}
 }
 
+//We operate on complex parametrized types here, makes sense to let compiler do the heavy typing excercises
+#[allow(clippy::default_trait_access)]
```

Comment:
maybe allow it in the whole module with `!`? Also, IMHO [it's](https://rust-lang.github.io/rust-clippy/v0.0.212/index.html#default_trait_access) not bad.